case_studies/galaxy_clustering/prior.py

```diff
@@ -0,0 +1,338 @@
+                while flag_reject:
+                    g1_size_samples[i][j] = gennorm.rvs(
+                        self.G1_beta, self.G1_loc, self.G1_scale, 1
+                    )[0]
+                    g2_size_samples[i][j] = gennorm.rvs(
+                        self.G2_beta, self.G2_loc, self.G2_scale, 1
+                    )[0]
+                    flag_large = g1_size_samples[i][j] ** 2 + g2_size_samples[i][j] ** 2 >= 1
+                    flag_g1_large = g1_size_samples[i][j] >= 0.8
+                    flag_g2_large = g2_size_samples[i][j] >= 0.8
+                    flag_reject = flag_large or flag_g1_large or flag_g2_large
+        return g1_size_samples, g2_size_samples
+
+    def galaxy_flux_ratio(self, size):
+        gmm_gal = self.gmm_gal
+        flux_logdiff, _ = gmm_gal.sample(size)
+        flux_logdiff = flux_logdiff[:][:, 1:]
+        flux_logdiff = np.clip(flux_logdiff, -2.76, 2.76)
+        flux_ratio = np.exp(flux_logdiff)
+        flux_prop = np.ones((flux_logdiff.shape[0], self.n_bands))
+        for band in range(self.reference_band - 1, -1, -1):
+            flux_prop[:, band] = flux_prop[:, band + 1] / flux_ratio[:, band]
+        for band in range(self.reference_band + 1, self.n_bands):
+            flux_prop[:, band] = flux_prop[:, band - 1] * flux_ratio[:, band - 1]
+        return flux_prop
+
+    def make_catalog(
+        self,
+        flux_samples,
+        t_size_samples,
+        g1_size_samples,
+        g2_size_samples,
+        geo_galaxy,
+        geo_galaxy_cluster,
+        galaxy_locs,
+        galaxy_locs_cluster,
+    ):
+        res = []
+        for i, _ in enumerate(flux_samples):
+            ratios = self.galaxy_flux_ratio(len(geo_galaxy_cluster[i]) + len(geo_galaxy[i]))
+            fluxes = np.array(flux_samples[i])[:, np.newaxis] * np.array(ratios)
+            mock_catalog = pd.DataFrame()
+            if geo_galaxy_cluster[i]:
+                mock_catalog["RA"] = np.append(
+                    np.array(geo_galaxy_cluster[i])[:, 0], np.array(geo_galaxy[i])[:, 0]
+                )
+                mock_catalog["DEC"] = np.append(
+                    np.array(geo_galaxy_cluster[i])[:, 1], np.array(geo_galaxy[i])[:, 1]
+                )
+                mock_catalog["X"] = np.append(
+                    np.array(galaxy_locs[i])[:, 0], np.array(galaxy_locs_cluster[i])[:, 0]
+                )
+                mock_catalog["Y"] = np.append(
+                    np.array(galaxy_locs[i])[:, 1], np.array(galaxy_locs_cluster[i])[:, 1]
+                )
+            else:
+                mock_catalog["RA"] = np.array(geo_galaxy[i])[:, 0]
+                mock_catalog["DEC"] = np.array(geo_galaxy[i])[:, 1]
+                mock_catalog["X"] = np.array(galaxy_locs[i])[:, 0]
+                mock_catalog["Y"] = np.array(galaxy_locs[i])[:, 1]
+            mock_catalog["FLUX_R"] = fluxes[:, 1]
+            mock_catalog["MAG_R"] = -2.5 * np.log10(mock_catalog["FLUX_R"]) + 30
```

Comment:
Does the mock catalog need both flux and mag fields? It seems like just one may be enough

Reply:
(If we have both, they could potentially disagree, which seems like a problem.)